modules/zone/zone.go

```diff
@@ -198,11 +201,11 @@ type GlobalLookupFactory struct {
 func (s *GlobalLookupFactory) AddFlags(f *flag.FlagSet) {
 	f.BoolVar(&s.Subfactory.IPv4Lookup, "ipv4-lookup", true, "perform A lookups for each server")
 	f.BoolVar(&s.Subfactory.IPv6Lookup, "ipv6-lookup", true, "perform AAAA record lookups for each server")
-	f.IntVar(&s.CacheSize, "cache-size", 100000, "number of records to store in cache of successes, preventing duplicate lookups")
 }
 
 func (s *GlobalLookupFactory) Initialize(c *zdns.GlobalConf) error {
 	s.GlobalConf = c
+	s.CacheSize = c.Threads * 100
```

Comment:
100 seems high to me. Can you check--- are there any domains with more than 10 name servers? I suspect that 25 would be more reasonable.